### ISSUE TITLE
Add option for `context.newline` fallback

### ISSUE THREAD

**Issue by MorevM** [stylelint/stylelint#9281]
### What is the problem you're trying to solve?

Good day,

I'm a maintainer of the [stylelint-stylistic](https://github.com/stylelint-stylistic/stylelint-stylistic) package. 
Our tests include rules that add linebreaks using `context.newline`, which is provided by Stylelint ([example](https://github.com/stylelint-stylistic/stylelint-stylistic/blob/f9dc38017dfd071299fa4f80851b9c1765c0f2e9/lib/rules/value-list-comma-newline-after/index.js#L85)).

The issue is that `context.newline` [falls back to node:os.EOL](https://github.com/stylelint/stylelint/blob/2e6808604b5ea1248e2cd1de4d39c2f5156333f5/lib/lintPostcssResult.mjs#L36) when the source code contains no linebreaks. This makes tests without linebreaks platform-sensitive, whereas we need deterministic output in tests.

### Example ([link](https://github.com/stylelint-stylistic/stylelint-stylistic/blob/f9dc38017dfd071299fa4f80851b9c1765c0f2e9/lib/rules/value-list-comma-newline-after/index.test.js#L55) if needed)

```ts
{
  code: `a { background-size: 0, 0; }`,
  fixed: `a { background-size: 0,\n 0; }`,
}
```

This works fine on Unix, but on Windows the fixed output becomes `a { background-size: 0,\r\n 0; }`, causing the test to fail:

```ts
{
  code:          `a { background-size: 0, 0; }`,
  fixed:         `a { background-size: 0,\n 0; }`,
  unixOutput:    `a { background-size: 0,\n 0; }`,  // matches
  windowsOutput: `a { background-size: 0,\r\n 0; }` // fails
}
```

We need a way to avoid using the system EOL in tests - tests should be platform-agnostic.

We are migrating from [stylelint-test-rule-node](https://github.com/stylelint/stylelint-test-rule-node) to my own [@morev/stylelint-testing-library](https://github.com/MorevM/stylelint-testing-library) to use Vitest as a test runner. I could introduce a configuration flag to enforce Unix linebreaks when the source contains none, but I believe this would be better handled at the Stylelint level.

### What solution would you like to see?

Add a configuration flag to control the fallback linebreak behavior when the source code contains no linebreaks. For example:

```ts
stylelint.lint({
  fallbackNewline: 'system', // 'unix' | 'windows'
});
```

This would allow tests to enforce a consistent linebreak style (e.g., 'unix') regardless of the underlying operating system, making test output deterministic and platform-agnostic.

What do you think?

**Comment by ybiquitous** [comment]
@MorevM Thanks for the report. At first glance, it doesn't seem very reasonable to add a new config solely for testing. 🤔 

What if you could specify `\n` instead of `context.newline` in the rule? E.g.,

```diff
-if (primary.startsWith(`always`)) afterValue = context.newline + afterValue
+if (primary.startsWith(`always`)) afterValue = '\n' + afterValue
```

Ref: https://github.com/stylelint-stylistic/stylelint-stylistic/blob/f9dc38017dfd071299fa4f80851b9c1765c0f2e9/lib/rules/value-list-comma-newline-after/index.js#L85

**Comment by MorevM** [comment]
@ybiquitous Thanks for the reply.

The first thing I tried was to wrap each rule in a closure to provide `context.newline: '\n'`. 
However, some of my tests explicitly check for `\r\n` (see [example](https://github.com/stylelint-stylistic/stylelint-stylistic/blob/f9dc38017dfd071299fa4f80851b9c1765c0f2e9/lib/rules/function-parentheses-newline-inside/index.test.js#L116)). Although I've never seen anyone intentionally use CRLF, formally this (as well as replacing `context.newline` to `\n` in the rule source) would be a breaking change, which I want to avoid.

This is essentially the same as asking you to "pin `context.newline` to `\n` in Stylelint core."

I understand why you wouldn't want to introduce a new option for such a niche use case mostly relevant for deprecated stylistic rules. I also think this is suboptimal... but I don't see any other good options. I cannot mock `node:os` reliably. Well, I could, but since it's a third-party dependency, that would mean processing Stylelint entirely through Vite, which roughly doubles test runtime.

Perhaps instead of a separate option, we could create an object with some niche settings, which would at least provide a room for future extensions like this and allow to keep niche settings grouped together:

```ts
stylelint.lint({
  nicheSettings: {
    fallbackNewline: 'system', // 'unix' | 'windows'
  }
})
```

---

Alternatively, I could rely on the `process.ENV.VITEST` environment variable in the rule code and implement these conditions there, but I think that's poor design.

Alternatively, I could add this option to my own testing library. That would solve my problem, but my library is neither the only one nor the official one - and someone else might potentially run into the same issue.

**Comment by jeddy3** [comment]
@MorevM Thanks for the request and discussion.

> Alternatively, I could add this option to my own testing library.

I suggest we go with that approach for now. If more people do run into the same issue later down the line, we can reconsider the alternatives.
